warehouse/internal/repo/staging.go

```diff
@@ -293,3 +302,70 @@ func (repo *StagingFiles) GetAfterID(ctx context.Context, sourceID, destinationI
 
 	return repo.parseRows(rows)
 }
+
+func (repo *StagingFiles) SetStatuses(ctx context.Context, ids []int64, status string) (err error) {
+	repo.init()
+
+	if len(ids) == 0 {
+		return fmt.Errorf("no staging files to update")
+	}
+
+	sqlStatement := `
+		UPDATE
+		` + stagingTableName + `
+		SET
+		  status = $1,
+		  updated_at = $2
+		WHERE
+		  id = ANY($3);
+`
+	result, err := repo.DB.Exec(sqlStatement, status, repo.Now(), pq.Array(ids))
+	if err != nil {
+		return fmt.Errorf("update ids status: %w", err)
+	}
+
+	rowsAffected, err := result.RowsAffected()
+	if err != nil {
+		return fmt.Errorf("rows affected: %w", err)
+	}
+	if rowsAffected != int64(len(ids)) {
+		return fmt.Errorf("not all rows were updated: %d != %d", rowsAffected, len(ids))
+	}
+
+	return
+}
+
+func (repo *StagingFiles) SetErrorStatus(ctx context.Context, stagingFileID int64, stageFileErr error) error {
+	repo.init()
+
+	sqlStatement := `
+		UPDATE
+		` + stagingTableName + `
+		SET
+			status = $1,
+			error = $2,
+			updated_at = $3
+		WHERE
+			id = $4;`
+
+	result, err := repo.DB.Exec(
+		sqlStatement,
+		warehouseutils.StagingFileFailedState,
```

Comment:
Let's move this constant inside this repo itself. wdyk?

````Go
const (
	StagingFileSucceededState = "succeeded"
	StagingFileFailedState    = "failed"
	StagingFileExecutingState = "executing"
	StagingFileAbortedState   = "aborted"
	StagingFileWaitingState   = "waiting"
)
````

Reply:
Is this only for the repository? I believe services / business logic make use of them as well.